### DISCUSSION TITLE
Parse patterns issue

### DISCUSSION THREAD

**Discussion by pavelpereverzev** [mozman/ezdxf#1210] [Issues] [answered]
Hello! I'm using ezdxf to convert spatial data (files like *.geojson) into dxf drawings. 
Recently got a new task to generate hatches with pattern fills. Pattern is actially a *.lin file made for styling polylines but also used for manual object styling.

I looked through manual and see this example: [https://ezdxf.readthedocs.io/en/stable/tutorials/hatch.html#load-hatch-patterns-from-file](https://ezdxf.readthedocs.io/en/stable/tutorials/hatch.html#load-hatch-patterns-from-file).
I can't get what actually `pattern` object/variable is from here:
`patterns = pattern.parse(EXAMPLE)`

I mean looks like `pattern` is not imported or generated here. I tried looking for this class in a root search but nothing came out.

**Comment by mozman** [comment]
The referenced example parses a `.pat` file. A `.lin` file contains linetypes which are different from hatch pattern.

![grafik](https://github.com/user-attachments/assets/8a08a88e-01ef-49e3-85dc-3d74ffc0496c)

**Reply by pavelpereverzev** [reply]
Ok, got it, thanks!
But what is `pattern` object/variable which parses `.pat` file contents?

**Reply by mozman** [reply, marked as answer]
```python
from ezdxf.tools import pattern
```

**Reply by pavelpereverzev** [reply]
Thanks!
